### PR TITLE
Support for namedtuples in array analysis

### PR DESCRIPTION
**Pull request by ehsantn**
Array analysis needs to support namedtuples since they work like tuples and their elements can be accessed.

### CODE REVIEW

File: numba/ir_utils.py [hunk 1]
```diff
@@ -1630,3 +1630,17 @@ def set_index_var_of_get_setitem(stmt, new_index):
...
+
+def is_namedtuple_class(c):
+    """check if c is a namedtuple class"""
+    if not isinstance(c, type):
+        return False
+    # should have only tuple as superclass
+    bases = c.__bases__
+    if len(bases) != 1 or bases[0] != tuple:
+        return False
+    # should have _fields that is all string
```
Comment (stuartarchibald) [inline numba/ir_utils.py]: Also has a `_make`, is this worth checking for?

Comment (ehsantn) [reply numba/ir_utils.py]: done.

**Comment by stuartarchibald** [comment]
Thanks for the fix @ehsantn, this can be merged once CI passes.

**Comment by seibert** [comment]
CI passed.  Merging

**Pull request merged**